### PR TITLE
Add `harvester-release.yaml` file to track versions

### PR DESCRIPTION
**Pull request by bk201**
Create a file to track component versions. e.g.,
```
harvester: f6bc010-dirty
harvesterChart: 0.0.0-dev
os: Harvester dfd4a9d-dirty
kubernetes: v1.21.5+rke2r1
rancher: v2.6-638b29a687f3818e7de676a573ec9b9156f193a9-head
monitoringChart: 100.0.0+up16.6.0
```

The file is stored in `/etc/harvester-release.yaml` and the release ISO.

- `harvester`: The harvester version. Formal release tag or commit hash of `harvester/harvester` project.
- `harvesterChart`: The harvester chart version, which is tracked in `harvester` ManagedChart.
- `os`: The installer version. The version part (without `Harvester ` prefix) is identical to `harvester` for a formal release. We record the version in `PRETTY_NAME` option in the os `/etc/os-release` file as well. 
   `PRETTY_NAME` is the only thing that can indicate the OS version in Kubernetes nodes. It's exposed in `.status.nodeInfo.osImage` of a node:
    ```
    # kubectl get node node1 -o yaml | yq e '.status.nodeInfo' -
    architecture: amd64
    bootID: 3f84cc62-5f0f-4105-9306-df7a56807b34
    containerRuntimeVersion: containerd://1.4.9-k3s1
    kernelVersion: 5.3.18-59.34-default
    kubeProxyVersion: v1.21.5+rke2r1
    kubeletVersion: v1.21.5+rke2r1
    machineID: 9a47446a1c3842e6c9bbaac661a60ff2
    operatingSystem: linux
    osImage: Harvester dfd4a9d-dirty               <--- PRETTY_NAME
    systemUUID: 2ffb9b8a-c6c7-4d65-9273-1a22da475e83
    ```
    We can use the info to track if a node's OS is upgraded.

- `rancher`: Embedded Rancher server version.
- `monitoringChart`: Monitoring chart version, which is tracked in `rancher-monitoring-crd` and `rancher-monitoring` ManagedCharts.

### CODE REVIEW

File: package/harvester-os/Dockerfile [hunk 1]
```diff
@@ -6,8 +6,8 @@ RUN curl -sfL https://github.com/rancher/wharfie/releases/latest/download/wharfi
...
 
-RUN rm -f /system/oem/05_network.yaml
+COPY harvester-release.yaml /etc/
 
-ARG HARVESTER_INSTALLER_VERSION
-
-RUN echo GRUB_ENTRY_NAME="\"Harvester $HARVESTER_INSTALLER_VERSION\"" >> /etc/os-release
+ARG HARVESTER_PRETTY_NAME
+RUN sed -i "s/^PRETTY_NAME.*/PRETTY_NAME=\"$HARVESTER_PRETTY_NAME\"/g" /etc/os-release
+RUN echo GRUB_ENTRY_NAME="\"$HARVESTER_PRETTY_NAME\"" >> /etc/os-release
```
Comment (guangbochen) [inline package/harvester-os/Dockerfile:13]: prefer to consolidate the line with `&&`

**Comment by gitlawr** [comment]
Any reason for using the yaml format? Personally speaking, I prefer to align it with `/etc/os-release`

**Comment by bk201** [comment]
> Any reason for using the yaml format? Personally speaking, I prefer to align it with `/etc/os-release`(or append those variables directly to `/etc/os-release`?)

It's easier to be parsed by `yq` in job scripts and unmarshalled in go codes.

File: package/harvester-os/Dockerfile (hunk 1 above)
Comment (bk201) [reply package/harvester-os/Dockerfile]: fixed!

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Pull request merged**